web/static/js/components/group_label_input.jsx

```diff
@@ -7,23 +7,63 @@ import styles from "./css_modules/group_label_input.css"
 // exceed the 1/4th column width on desktop, even with the widest possible characters
 const MAX_LENGTH_OF_GROUP_NAME = 20
 
-const GroupLabelInput = ({ groupWithAssociatedIdeasAndVotes, actions }) => (
-  <div className="ui fluid input">
-    <input
-      type="text"
-      className={styles.textInput}
-      placeholder="Add a group label"
-      defaultValue={groupWithAssociatedIdeasAndVotes.label}
-      maxLength={MAX_LENGTH_OF_GROUP_NAME}
-      onBlur={e => {
-        actions.submitGroupLabelChanges(groupWithAssociatedIdeasAndVotes, e.target.value)
-      }}
-    />
-    <div className="instruction">
-      <span className="keyboard-key">tab</span> to submit
-    </div>
-  </div>
-)
+class GroupLabelInput extends Component {
+  state = (() => {
+    const { groupWithAssociatedIdeasAndVotes } = this.props
+    const { label } = groupWithAssociatedIdeasAndVotes
+    return {
+      oldLabel: label,
+      showCheckmarkForLabelId: null,
+    }
+  })();
+
+  static getDerivedStateFromProps(props, state) {
+    const { oldLabel } = state
+    const { groupWithAssociatedIdeasAndVotes } = props
+    const newLabel = groupWithAssociatedIdeasAndVotes.label
+    if (oldLabel !== newLabel) {
+      return ({
+        oldLabel: newLabel,
+        showCheckmarkForLabelId: groupWithAssociatedIdeasAndVotes.id })
+    }
+    return null
+  }
+
+  labelUpdatedCheckmark = () => {
+    const classes = ["check icon"]
+    classes.push(styles.updateSucceededCheckmark)
+    setTimeout(() => {
```

Comment:
i think there's a bug here, in that this `labelUpdatedCheckmark ` function could fire multiple times due to the react render cycle firing for reasons outside this particular component's control, i.e. a parent component's re-renders could cause this component to re-render multiple times a second, causing this function to be re-invoked multiple times, causing potential race conditions with the number of `setTimeouts` that get run

Reply:
Thanks for letting me know of this issue.  I am not sure how to properly fix the problem.  But, I'd love to pair with you as per your other comment to resolve this.

Reply:
We addressed this as an unlikely scenario, but may merit investigation in the future.